File: doc/codes/python/examples/api/basics/client-config.py
import asyncio

import autd3_link_nop as nop
from autd3 import Client, ClientConfig, MAX_INFLIGHT
from autd3.geometry import Autd3, Geometry


async def main() -> None:
    geometry = Geometry([Autd3([0.0, 0.0, 0.0], [1.0, 0.0, 0.0, 0.0])])

    link = nop.Nop()
    option = (
        # ANCHOR: config
        ClientConfig(
            timeout_cycles=10,
            max_inflight=MAX_INFLIGHT,
            max_resync_rounds=8,
            low_latency=False,
            reset_resend_cycles=2,
            rt_priority=None,
            rt_affinity=None,
            validate_state=True,
        )
        # ANCHOR_END: config
    )
    # ANCHOR: api
    await Client.open(geometry, link, option)
    # ANCHOR_END: api


asyncio.run(main())

File: doc/codes/rust/examples/api/basics/client-config.rs
use anyhow::Result;

use std::num::{NonZeroU32, NonZeroUsize};

use autd3_rs::geometry::{Autd3, Geometry};
use autd3_rs::{
    Client, ClientConfig, MAX_INFLIGHT, RtSchedulePolicy, ThreadPriority, ThreadPriorityValue,
};
use autd3_rs_link_nop::Nop;

#[tokio::main(flavor = "multi_thread")]
async fn main() -> Result<()> {
    let geometry = Geometry::new(vec![Autd3::default()]);

    let link = Nop;
    let option =
        // ANCHOR: config
        ClientConfig {
            timeout_cycles: NonZeroU32::new(10).unwrap(),
            max_inflight: NonZeroUsize::new(MAX_INFLIGHT).unwrap(),
            max_resync_rounds: NonZeroU32::new(8).unwrap(),
            low_latency: false,
            reset_resend_cycles: NonZeroU32::new(2).unwrap(),
            rt_priority: Some(ThreadPriority::Crossplatform(
                ThreadPriorityValue::try_from(80).unwrap(),
            )),
            rt_policy: RtSchedulePolicy::Fifo,
            rt_affinity: None,
            validate_state: true,
            require_supported_firmware: false,
            ..Default::default()
        }
        // ANCHOR_END: config
        ;
    // ANCHOR: api
    Client::open(&geometry, link, option).await?;
    // ANCHOR_END: api

    Ok(())
}
